app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,72 @@
+class TasksController < ApplicationController
+  before_action :set_task, only: [:destroy, :update, :show, :edit]
+  before_action :set_event
+
+  def new
+    @task = Task.new
+  end
+
+  def index
+    @task = Task.all_tasks(@event.id)
+    if !@task
+      redirect_to new_event_task_url
+    end
+  end
+
+  def show
+    if !@task
+      redirect_to new_event_task_url
+    end
+  end
+
+  def edit
+  end
+
+  def create
+    some = params["task"]
+    some["event_id"] = @event.id
```

Comment:
this whole "some" thing is super obtuse and hard to read, try to refactor using some built in methods.